prog/vm/nexus.rb

```diff
@@ -340,21 +340,17 @@ def before_run
       create_storage_volume_records(vm_host, storage_volumes)
     end
 
-    Clog.emit("vm allocated") do
-      {allocation: {vm_host_ubid: vm_host.ubid, ip4: ip4, address: address&.cidr&.to_s,
-                    routed_to: if (ahid = address&.routed_to_host_id)
-                                 UBID.from_uuidish(ahid).to_s
-                               end}}
-    end
     fail "no ip4 addresses left" if vm.ip4_enabled && !ip4
 
     DB.transaction do
       vm.update(
         vm_host_id: vm_host_id,
         ephemeral_net6: vm_host.ip6_random_vm_network.to_s,
-        local_vetho_ip: vm_host.veth_pair_random_ip4_addr.to_s
+        local_vetho_ip: vm_host.veth_pair_random_ip4_addr.to_s,
+        allocated_at: Time.now
       )
 
+      Clog.emit("vm allocated") { {vm: vm.values, allocation: {vm_ubid: vm.ubid, vm_host_ubid: vm_host.ubid, duration: Time.now - vm.created_at}} }
```

Comment:
Looks like this drops routed_to, which is fine, but kinda not a great thing to slide into a patch on the side (if being very particular, break it out and maybe say why we want to lose that field first)

Reply:
Yeah. it's not good to change it silently. I moved it to a separate commit.

```
commit 658270dad4c53a77a39917056bacaf4a74d90f39
Author: Enes Cakir <enes@ubicloud.com>
Date:   Fri Mar 22 14:44:01 2024 +0300

    Refactor vm allocation log

    The IPv4 allocation follows this line. Even if we print this, we may not
    allocate a VM on this host. It's best to print when the allocation is
    finalized.

    In the past, we encountered concurrency and lease violation problems
    with respirate, resulting in the same IP being assigned to multiple VMs.
    We included detailed information in the log to assist with debugging.
    However, these values have become too verbose and are no longer useful.
```